src/responses/search-request-params.ts

```diff
@@ -0,0 +1,12 @@
+export interface SearchRequestParams {
+  client?: string;
+  user_query?: string;
+  service_backend?: string;
+  page_type?: string;
+  page?: number;
+  hits_per_page?: number;
+  fields?: string[];
+  sort?: string[];
+  aggregations?: string[];
+  aggregations_size?: number;
```

Comment:
this shape looks familiar...like, `SearchParams` - what is the difference between the two?

Reply:
This `SearchRequestParams` is the "normalized" request parameters as provided in the PPS response, whereas `SearchParams` is the interface that callers can provide their params as.

While many of the property names do overlap, I think the important difference is that `SearchParams` enables things like sorting and aggregations to be specified in a more structured way, like:
```js
params: {
  sort: [
    {
      field: 'title',
      direction: 'asc'
    },
    {
      field: 'date',
      direction: 'desc'
    }
  ]
}
```

...whereas the `SearchRequestParams` holds the raw, parsed values that have been provided directly to the server
(like `sort: ['title:asc', 'date:desc']`).

(And similarly for structured aggregations etc.)

Reply:
For clarity's sake, the PPS responses are generally structured like this (showing the rough mapping onto SearchService model types):

```js
{
  request: {
    client_parameters: SearchRequestParams,
    finalized_parameters: SearchRequestParams
  },
  elapsed_secs: number,
  response: {
    header: SearchResponseHeader,
    body: SearchResponseDetails,
    hit_schema: SearchHitSchema
  }
}
```